packages/app/features/activity/screen.tsx

```diff
@@ -132,65 +130,170 @@ function ActivityBody() {
+                    key={item.send_id}
+                    keyField={keyField}
+                    item={item}
+                    query={query}
+                  />
+                ))}
+              </XStack>
+            </YStack>
+          ) : null
         )}
-      {['send_id_matches', 'tag_matches', 'phone_matches'].map(
-        (key) =>
-          results[key] &&
-          results[key].length > 0 && (
-            <div key={key}>
-              {' '}
-              {/* Use a div with a key instead of a fragment */}
-              <H4 theme={'alt2'}>{key.replace('_', ' ')}</H4>
-              {results[key].map((result) => (
-                <Link key={result.send_id} href={`/profile/${result.send_id}`}>
-                  <XStack testID={`tag-search-${result.send_id}`} ai="center" space="$4">
-                    <Avatar size="$4" br="$4" space="$2">
-                      <Avatar.Image src={result.avatar_url} />
-                      <Avatar.Fallback>
-                        <Avatar>
-                          <Avatar.Image
-                            src={`https://ui-avatars.com/api.jpg?name=${result.send_id}&size=256`}
-                          />
-                          <Avatar.Fallback>
-                            <Paragraph>??</Paragraph>
-                          </Avatar.Fallback>
-                        </Avatar>
-                      </Avatar.Fallback>
-                    </Avatar>
-                    <YStack space="$1">
-                      <Text>{result.tag_name ? result.tag_name : result.send_id}</Text>
-                    </YStack>
-                  </XStack>
-                </Link>
-              ))}
-            </div>
-          )
+      </YStack>
+    </Container>
+  )
+}
+
+function HighlightMatchingText({ text, highlight }: { text: string; highlight: string }) {
+  const regex = new RegExp(`(${highlight})`, 'gi')
+  const parts = text.split(regex)
+
+  return (
+    <Text
+      fontWeight={'300'}
+      $theme-light={{ color: '$darkGrayTextField' }}
+      $theme-dark={{ color: '$lightGrayTextField' }}
+      fontSize="$7"
+      $gtSm={{ fontSize: '$5' }}
+    >
+      {parts.map((part, i) =>
+        regex.test(part) ? (
+          <Text
+            key={i + part}
+            fontWeight="500"
+            $theme-light={{ color: '$black' }}
+            $theme-dark={{ color: '$white' }}
+          >
+            {part}
+          </Text>
+        ) : (
+          part
+        )
       )}
-    </YStack>
+    </Text>
   )
 }
+
+function SearchResultRow({ keyField, item, query }) {
+  const { resolvedTheme } = useThemeSetting()
+  const rowBC = resolvedTheme?.startsWith('dark') ? '$metalTouch' : '$gray2Light'
+
+  return (
+    <View
+      br="$5"
+      key={item.send_id}
+      width="100%"
+      $gtLg={{
+        width: 'calc((100% - 48px) / 3)',
+        bc: rowBC,
+        p: '$1.5',
+      }}
+      $gtXl={{
+        width: 'calc((100% - 72px) / 4)',
```

Comment:
sadly, we can't do this on native. not sure what the correct usage would be on native to match the style though.

```suggestion
        width: isWeb ? 'calc((100% - 48px) / 3)' : '100%',
```

Reply:
luckily it's just the 100% on mobile, so i'll just add that ternary in there